sqlglot/expressions.py

```diff
@@ -2055,6 +2056,23 @@ class Credentials(Expression):
     }
 
 
+class DistributedByHash(Expression):
+    arg_types = {"expressions": True, "buckets": False, "sorted_by": False}
+
+    @property
+    def auto_bucket(self) -> bool:
+        return self.args.get("buckets") is None
```

Comment:
These new expressions should all inherit from `Property` and be named accordingly right? See how other properties are handled.

Reply:
Got it